src/shared_modules/utils/zlibHelper.hpp

```diff
@@ -88,6 +92,89 @@ namespace Utils
             } while (len == sizeof(buf));
             outputFile.close();
         }
+
+        /**
+         * @brief Uncompress ZIP file and returns a list with the decompressed files.
+         *
+         * @param zipFilePath Compressed (.zip) file path.
+         * @param outputDir Folder where the output files will be stored.
+         * @return std::vector<std::string> List of decompressed files.
+         */
+        static std::vector<std::string> zipDecompress(const std::filesystem::path& zipFilePath,
+                                                      const std::filesystem::path& outputDir)
+        {
+            // Open .zip file.
+            auto unzFile {unzOpen(zipFilePath.c_str())};
+            if (!unzFile)
+            {
+                throw std::runtime_error {"Unable to open compressed file: " + zipFilePath.string()};
+            }
+
+            // Get .zip file information (amount of files, i.e.).
+            unz_global_info globalInfo;
+            if (unzGetGlobalInfo(unzFile, &globalInfo) != UNZ_OK)
+            {
+                unzClose(unzFile);
+                throw std::runtime_error {"Unable to get global information of file: " + zipFilePath.string()};
+            }
+
+            // Iterate all compressed files within the .zip file.
+            std::vector<std::string> decompressedFiles(globalInfo.number_entry);
+            for (uLong currentFileIndex {0}; currentFileIndex < globalInfo.number_entry; ++currentFileIndex)
+            {
+                unz_file_info fileInfo;
+                char filename[256];
+                if (unzGetCurrentFileInfo(unzFile, &fileInfo, filename, sizeof(filename), nullptr, 0, nullptr, 0) !=
+                    UNZ_OK)
+                {
+                    unzClose(unzFile);
+                    throw std::runtime_error {"Unable to get current file information"};
+                }
+
+                // Open current file within the .zip file.
+                if (unzOpenCurrentFile(unzFile) != UNZ_OK)
+                {
+                    unzClose(unzFile);
+                    throw std::runtime_error {"Unable to open current file: " + std::string(filename)};
+                }
+
+                // Read current file content.
+                std::vector<char> buffer(fileInfo.uncompressed_size);
+                if (unzReadCurrentFile(unzFile, buffer.data(), buffer.size()) < 0)
+                {
+                    unzCloseCurrentFile(unzFile);
+                    unzClose(unzFile);
+                    throw std::runtime_error {"Unable to read content of current file: " + std::string(filename)};
+                }
```

Comment:
We should set a buffer with a fixed size, and read the uncompressed data in chunks. That way we can control the max memory used. At least in case of 'big' files.

Reply:
Great idea. Done!

I defined the chunk size as equal to 100MB as it was the fastest when decompressing the snapshots (see PR description) and is not that great. Anyway, this is opened to discussion.